concent_api/verifier/utils.py

```diff
@@ -52,30 +52,41 @@ def prepare_storage_request_headers(file_transfer_token: message.FileTransferTok
 
 
 def store_file_from_response_in_chunks(response: requests.Response, file_path: str):
-    with open(file_path, 'x') as f:
+    with open(file_path, 'wb') as f:
         for chunk in response.iter_content():
             f.write(chunk)
 
 
-def run_blender(scene_file, output_format, script_file=''):
+def run_blender(scene_file, output_format, frame_number, script_file=''):
+    output_format = adjust_format_name(output_format)
     return subprocess.run(
         [
             "blender",
-            "-b", f"{scene_file}",
+            "-b", f"{settings.VERIFIER_STORAGE_PATH}/{scene_file}",
             "-y",  # enable scripting by default
             "-P", f"{script_file}",
-            "-o", generate_blender_output_file_name(scene_file),
+            "-o", f"{generate_base_blender_output_file_name(scene_file)}",
             "-noaudio",
-            "-F", f"{output_format.upper()}",
+            "-F", f"{output_format}",
             "-t", "1",  # cpu_count
-            "-f", "1",  # frame
+            "-f", f"{frame_number}",  # frame
         ],
         timeout=settings.BLENDER_MAX_RENDERING_TIME,
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
     )
 
 
+def adjust_format_name(output_format: str):
+    """
+    This function enforces the upper case for format name.
+    For desired JPG format, the parameter for blender should be JPEG and the extension of result file is *.jpg.
+    """
+    if output_format.upper() == 'JPG':
+        return 'JPEG'
+    return output_format.upper()
+
+
 def unpack_archive(file_path):
     """ Unpacks archive in chunks. """
     with zipfile.ZipFile(os.path.join(settings.VERIFIER_STORAGE_PATH, file_path), 'r') as zf:
```

Comment:
You can also change this short variable name `zf` :)

Reply:
done